fluent-react/src/localization.ts

```diff
@@ -73,6 +79,195 @@ export class ReactLocalization {
     return fallback || id;
   }
 
+  getFragment(
+    id: string,
+    args?: {
+      vars?: Record<string, FluentVariable>,
+      elems?: Record<string, ReactElement>,
+    },
+    fallback?: string
+  ): ReactFragment {
+    return this.getElement(createElement(Fragment, null, fallback || id), id, args);
+  }
+
+  getElement(
+    component: ReactNode | Array<ReactNode>,
+    id: string,
+    args?: {
+      vars?: Record<string, FluentVariable>,
+      elems?: Record<string, ReactElement>,
+      attrs?: Record<string, boolean>;
+    },
+  ): ReactElement {
+    let componentToRender: ReactNode | null;
+
+    // Validate that the child element isn't an array that contains multiple
+    // elements.
+    if (Array.isArray(component)) {
+      if (component.length > 1) {
+        throw new Error(
+          "<Localized/> expected to receive a single React node child"
+        );
+      }
+
+      // If it's an array with zero or one element, we can directly get the first
+      // one.
+      componentToRender = component[0];
+    } else {
+      componentToRender = component ?? null;
+    }
+
+    const bundle = this.getBundle(id);
+    if (bundle === null) {
+      if (id === undefined) {
+        this.reportError(
+          new Error("No string id was provided when localizing a component.")
+        );
+      } else {
+        if (this.areBundlesEmpty()) {
+          this.reportError(
+            new Error(
+              "Attempting to get a localized element when no localization bundles are " +
+                "present."
+            )
+          );
+        } else {
+          this.reportError(
+            new Error(
+              `The id "${id}" did not match any messages in the localization ` +
+                "bundles."
+            )
+          );
+        }
+      }
+      return createElement(Fragment, null, componentToRender);
+    }
+
+    // this.getBundle makes the bundle.hasMessage check which ensures that
+    // bundle.getMessage returns an existing message.
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const msg = bundle.getMessage(id)!;
+
+    let errors: Array<Error> = [];
+
+    // Check if the component to render is a valid element -- if not, then
+    // it's either null or a simple fallback string. No need to localize the
+    // attributes.
+    if (!isValidElement(componentToRender)) {
```

Comment:
Hmm, I'm not sure what the best way to do that is - there'd be quite a bit of duplication in initialising the bundle and handling its various error cases. Personally I wouldn't be too worried about doing _too much_ error handling. What I could do is update the type definition of `getElement` to only allow `ReactChild | ReactFragment`? Then we're technically requiring it to be a valid element in the public API.

Or, hmm, what I could also do is update `<Localized>` to check for non-element children and, if found, wrap them in a fragment and pass that to `getElement`? That would cause the latter to do some unnecessary work, but nothing too egregious I thinkg - what do you think?

Reply:
How about calling `getString()` rather than `getElement()` from a Localized with non-element children, and wrapping the result in a Fragment? As far as I can tell, that should replicate the current behaviour.

Reply:
Ah yeah, that's clever: 59671ccd

I should note that that does change the behaviour a little bit: if you do not provide a child, it will now render the string ID, rather than `null`. Which makes sense to me and aligns it with the existing behaviour of `getString()`, but I can also imagine that that's not desirable.

One way to work around that might be to set the fallback string to some unique string, and return `null` instead of a fragment if `getString` returns the fallback string, but that feels pretty hacky :/ Alternatively, although still a change in behaviour but closer to the previous behaviour, would be to set the fallback string to the empty string, resulting in it returning a fragment containing an empty string instead of `null`. That's technically different from what it used to be, but should have the same result for the user.